src/awkward/numba.py

```diff
@@ -2,37 +2,37 @@
 
 import awkward as ak
 
-checked_version = False
+_has_checked_version = False
 
 
 def register_and_check():
-    global checked_version
+    global _has_checked_version
 
-    if not checked_version:
-        try:
-            import numba
-        except ImportError as err:
-            raise ImportError(
-                """install the 'numba' package with:
+    try:
+        import numba
+    except ImportError as err:
+        raise ImportError(
+            """install the 'numba' package with:
 
-    pip install numba --upgrade
+pip install numba --upgrade
 
 or
 
-    conda install numba"""
-            ) from err
+conda install numba"""
+        ) from err
 
-        checked_version = True
+    if not _has_checked_version:
         if ak._util.parse_version(numba.__version__) < ak._util.parse_version("0.50"):
             raise ImportError(
                 "Awkward Array can only work with numba 0.50 or later "
                 "(you have version {})".format(numba.__version__)
             )
+        _has_checked_version = True
 
-    register()
+    _register()
 
 
-def register():
+def _register():
```

Comment:
The motivation for this being a public function was so that users could call

```python
ak.numba.register()
```

if the entry-point didn't work for some reason. That's a backdoor, but it can be an important one because the entry-point mechanism depends on issues in the environment being set up correctly and that's distinct from code. (Sometimes, I've had valid code, incorrectly installed, and puzzled over why something that should be in the entry-point wasn't.)

I'd like to decouple the _possibility_ of getting Awkward-Numba to work from the messy world of people not following the best practices for package installation. (You know, that XKCD.) When somebody reports a Numba error message saying that the ak.Array type is not recognized, I'd like to be able to say, "Fix you environment; but for now, use `ak.numba.register()`."

So it would be good for this function to be public. (Otherwise, why have an `ak.numba` module at all?)

Reply:
Oh, that's true—if an external user is calling this, we don't want it to go by without ensuring that the Numba version is acceptable. (That's the problem with soft-imports!)

If we're going to have multiple third-party registration functions, presenting them all with the same name would make it easier for users to guess—fewer round-trips to the documentation. On the other hand, if they all start with "register*", then tab-completion would resolve it, too.

Reply:
Or... maybe they're _all_ named `register_and_check`. These two are.